traitlets/traitlets.py

```diff
@@ -390,20 +385,18 @@ def _setup_dynamic_initializer(self, obj):
         # trait declaration or above.
         mro = type(obj).mro()
         meth_name = '_%s_default' % self.name
-        for cls in mro[:mro.index(self.this_class)+1]:
+        for cls in mro[:mro.index(self.this_class) + 1]:
             if meth_name in cls.__dict__:
                 break
         else:
-            return False
-        # Complete the dynamic initialization.
+            return
         obj._trait_dyn_inits[self.name] = meth_name
-        return True
 
     def _set_default_value_at_instance_init(self, obj):
         # As above, but if no default was specified, don't try to set it.
         # If the trait is accessed before it is given a value, init_default_value
         # will be called at that point.
-        if (not self._setup_dynamic_initializer(obj)) \
+        if (self.name not in obj._trait_dyn_inits) \
```

Comment:
I dislike this change - it makes the API more stateful, because `_set_default_value_at_instance_init()` now relies on `_setup_dynamic_initializer` having been called first. But in #23, you remove this method entirely. The two PRs will conflict - what is the relationship between them?